### ISSUE TITLE
Parcel Incorrectly Detects Destructuring Assignment As Module Usage

### ISSUE THREAD

**Issue by an0nusr** [parcel-bundler/parcel#7597]
# 🐛 bug report

When using Parcel 2.2.0 with a single HTML file with an inline script, Parcel incorrectly identifies a destructuring assignment as some sort of module, triggering the error: `@parcel/transformer-js: Browser scripts cannot have imports or exports.` when running `parcel build`.

Writing this will trigger the error in an inline script block:
```js
let [prev, last] = getLastLine(currentBlock)
```

However, writing this does not:
```js
let res = getLastLine(currentBlock)
let prev = res[0]
let last = res[1]
```

## 🎛 Configuration (.babelrc, package.json, cli command)

Package json:
```js
{
  "name": "furryfont",
  "version": "1.0.0",
  "source": "index.html",
  "browserslist": "> 0.5%, last 2 versions, not dead",
  "author": "anonusr",
  "license": "AGPL-3.0-only",
  "scripts": {
    "start": "parcel",
    "build": "parcel build",
    "deploy": "echo 'furryfont.anonusr.org' > ./dist/CNAME && gh-pages -d dist"
  },
  "devDependencies": {
    "gh-pages": "^3.2.3",
    "parcel": "^2.2.0"
  }
}
```

## 🤔 Expected Behavior

The inline script, that does not use any modules, should build as expected.

## 😯 Current Behavior

Parcel incorrectly declares that I'm using a module when I attempt to build the package:

```
🚨 Build failed.

@parcel/transformer-js: Browser scripts cannot have imports or exports.

  /home/user/furryfont/index.html:81:1
    80 |
  > 81 |         <script>
  >    | ^
    82 |             function render_editor() {
    83 |                 window.quill = new Quill('#editor', {

  💡 Add the type="module" attribute to the <script> tag.
  📝 Learn more: https://parceljs.org/languages/javascript/#classic-scripts

error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

## 🔦 Context

This was frustrating as I needed to go through my last commit line by line to see what was causing the build system to fail. I knew I hadn't used an import, so the error caught me offguard. 

## 💻 Code Sample
You can view my code here: https://github.com/an0nusr/furryfont/commits/main

The issue began when I introduced the destructuring assignment in commit 9df206bc65b559d8326f5263d66f3b4e1cbdac23.

It resolved when I rewrote the line in commit: 7e279a691c8faeeaac536debf3557b1ec838322c

## 🌍 Your Environment

| Software         | Version(s) |
| ---------------- | ---------- |
| Parcel           | 2.2.0 |
| Node             | v12.22.7 | 
| npm/Yarn         | yarn 1.22.5 |
| Operating System | Ubuntu 18.04 |

**Comment by Shinyaigeek** [comment]
This bug seems to be caused by [this line](https://github.com/parcel-bundler/parcel/blob/v2/packages/transformers/js/core/src/dependency_collector.rs#L246). when `swc` parses code, `swc` replaces `let [prev, last] = getLines()` with 

```javascript
const _helpers = require("@swc/helpers");
let [prev, last] = _helpers.slicedToArray(getLastLine());
```

**Comment by Shinyaigeek** [comment]
I sent PR to fix this #7599
